core/templates/pages/blog-dashboard-page/navbar/navbar-breadcrumb/blog-dashboard-navbar-breadcrumb.component.spec.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the blog dashboard navbar breadcrumb.
+ */
+
+import { ComponentFixture, TestBed, fakeAsync, waitForAsync, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { BlogDashboardNavbarBreadcrumbComponent } from 'pages/blog-dashboard-page/navbar/navbar-breadcrumb/blog-dashboard-navbar-breadcrumb.component';
+import { BlogDashboardPageService } from 'pages/blog-dashboard-page/services/blog-dashboard-page.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { EventEmitter } from '@angular/core';
+
+
+describe('Blog Dashboard Page Component', () => {
+  let blogDashboardPageService: BlogDashboardPageService;
+  let component: BlogDashboardNavbarBreadcrumbComponent;
+  let fixture: ComponentFixture<BlogDashboardNavbarBreadcrumbComponent>;
+  let mockWindowRef: MockWindowRef;
+
+  class MockWindowRef {
+    nativeWindow = {
+      location: {
+        href: '',
+        hash: '/'
+      },
+      open: (url) => { },
+      onhashchange() {
+        return this.location._hashChange;
+      },
+    };
+  }
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+      ],
+      declarations: [
+        BlogDashboardNavbarBreadcrumbComponent,
+      ],
+      providers: [
+        BlogDashboardPageService,
+        {
+          provide: WindowRef,
+          useClass: MockWindowRef
+        },
+      ],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      BlogDashboardNavbarBreadcrumbComponent);
+    component = fixture.componentInstance;
+    mockWindowRef = TestBed.inject(WindowRef) as unknown as MockWindowRef;
+    blogDashboardPageService = TestBed.inject(BlogDashboardPageService);
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+
+  it('should create', () => {
+    expect(component).toBeDefined();
+  });
+
+  it('should intialize and set active tab according to url', fakeAsync(() => {
+    component.ngOnInit();
+    expect(component.activeTab).toBe('main');
+
+    blogDashboardPageService.navigateToEditorTabWithId('123456sample');
+    mockWindowRef.nativeWindow.onhashchange();
+    tick();
```

Comment:
Why do we need `tick()` here?

Reply:
To detect the url change in blog dashboard page service after the hash change event is emitted:
![Screenshot from 2021-08-14 10-33-39](https://user-images.githubusercontent.com/68547101/129434722-3b3afee9-256f-49d1-9ebd-0e66fe690e53.png)